code/game/turfs/simulated/floor/plating/lava.dm

```diff
@@ -54,10 +54,14 @@
 				O.armor["fire"] = 50
 			O.fire_act(10000, 1000)
 
-
 		else if (isliving(thing))
 			. = 1
 			var/mob/living/L = thing
+			if(ishuman(thing))
```

Comment:
This whole section I'm not a big fan of.

Is there no existing var or flag we can override? If not, I'd prefer if you set one up Then simply set the flag on the mob when the flightpack is equipped and then check that. I can forsee other potential items that may want to set things flying

I know there is  a FLYING  flag in species.specflags that lava and chasm's check, or the simple animal.flying var.

So actually in fact, move the simple_animal FLYING var up to the mob level and set that, then update the existing code checks in chasm and lava for this (change the istype check to ismob instead of issimpleanimal)

You could possibly refactor things by turning the flying var into movement_type and making it a bit flag, so that people could implement different types of moving, or turn multiple types of movement on and off within the one flag, but that significant of a refactor is I think a little too much to ask, so just the flying var up to mob level will be fine.